### DISCUSSION TITLE
Fourth-party dependencies

### DISCUSSION THREAD

**Discussion by Saewo0** [open-telemetry/opentelemetry-java#3187] [Q&A] [answered]
Hi, I am using OpenTelemetry api & sdk as third-party libs for my project. Does anyone know about how to find the fourth-party dependencies used by OT api & sdk?

I have tried the command `gradlew submodule:dependencoes --configuration compileClasspath`. According to the result as shown below, it seems there's no external references to any fourth-party libs.
![image](https://user-images.githubusercontent.com/29158720/116487965-1812e280-a846-11eb-9579-e27fff17b59c.png)
I can tell that "io.opentelemetry:opentelemetry-api" depends on "io.opentelemetry:opentelemetry-context".

However, I do see some external dependencies in "io.opentelemetry:opentelemetry-context", like "io.grpc.grpc-context" and "io.zipkin.brave". See here (https://github.com/open-telemetry/opentelemetry-java/blob/7c9ea8b0f6340513908d188eb0d2fb1f4712d6fa/context/build.gradle.kts). Am I misunderstanding something?

Thanks in advance!

**Comment by jkwatson** [comment, marked as answer]
We intentionally have no external dependencies for the core API and sdk. The dependencies you see for the context are test dependencies only.
